tests/lean/run/async_http_uri_parser.lean

```diff
@@ -0,0 +1,327 @@
+
+/--
+info: Std.Http.RequestTarget.absoluteForm
+  { scheme := "http",
+    authority := some { userInfo := none, host := Std.Http.URI.Host.ipv6 2001:db8::1, port := some 8080 },
+    path := { segments := #["path"], absolute := true },
+    query := none,
+    fragment := none }
+-/
+#guard_msgs in
+#eval show IO _ from do
+  let result ← runParser Std.Http.Parser.parseRequestTarget "http://[2001:db8::1]:8080/path"
+  IO.println (repr result)
+
+/--
+info: Std.Http.RequestTarget.absoluteForm
+  { scheme := "https",
+    authority := some { userInfo := none, host := Std.Http.URI.Host.name "example.com", port := none },
+    path := { segments := #["page"], absolute := true },
+    query := none,
+    fragment := some "section1" }
+-/
+#guard_msgs in
+#eval show IO _ from do
+  let result ← runParser Std.Http.Parser.parseRequestTarget "https://example.com/page#section1"
+  IO.println (repr result)
+
+/--
+error: offset 0: invalid request target
+-/
+#guard_msgs in
+#eval show IO _ from do
+  let result ← runParser Std.Http.Parser.parseRequestTarget "?query=only"
+  IO.println (repr result)
+
+/--
+error: offset 1: it's a scheme starter
+-/
+#guard_msgs in
+#eval show IO _ from do
+  let result ← runParser Std.Http.Parser.parseRequestTarget "//double//slash//path"
+  IO.println (repr result)
+
+/--
+info: Std.Http.RequestTarget.originForm
+  { segments := #["very", "long", "path", "with", "many", "segments", "and", "encoded spaces", "and+plus+signs",
+                  "final/segment"],
+    absolute := true }
+  none
+  none
+-/
+#guard_msgs in
+#eval show IO _ from do
+  let result ← runParser Std.Http.Parser.parseRequestTarget "/very/long/path/with/many/segments/and/encoded%20spaces/and%2Bplus%2Bsigns/final%2Fsegment"
+  IO.println (repr result)
+
+/--
+info: Std.Http.RequestTarget.originForm
+  { segments := #["api"], absolute := true }
+  (some #[("filters[]", some "active"), ("filters[]", some "verified"), ("sort[name]", some "asc"),
+     ("sort[date]", some "desc")])
+  none
+-/
+#guard_msgs in
+#eval show IO _ from do
+  let result ← runParser Std.Http.Parser.parseRequestTarget "/api?filters%5B%5D=active&filters%5B%5D=verified&sort%5Bname%5D=asc&sort%5Bdate%5D=desc"
+  IO.println (repr result)
+
+/--
+info: Std.Http.RequestTarget.absoluteForm
+  { scheme := "https",
+    authority := some { userInfo := none, host := Std.Http.URI.Host.name "xn--nxasmq6b.xn--o3cw4h", port := none },
+    path := { segments := #["path"], absolute := true },
+    query := none,
+    fragment := none }
+-/
+#guard_msgs in
+#eval show IO _ from do
+  let result ← runParser Std.Http.Parser.parseRequestTarget "https://xn--nxasmq6b.xn--o3cw4h/path"
+  IO.println (repr result)
+
+/--
+info: Std.Http.RequestTarget.authorityForm
+  { userInfo := none, host := Std.Http.URI.Host.name "localhost", port := some 65535 }
+-/
+#guard_msgs in
+#eval show IO _ from do
+  let result ← runParser Std.Http.Parser.parseRequestTarget "localhost:65535"
+  IO.println (repr result)
```

Comment:
Here is a failing test (actually a regression in one of the previous few commits):
```lean
/-- info: Std.Http.RequestTarget.originForm { segments := #[], absolute := true } none none -/
#guard_msgs in
#eval show IO _ from do
  let result ← runParser Std.Http.Parser.parseRequestTarget "/"
  IO.println (repr result)
```